pkg/operator/deploy/deploy.go

```diff
@@ -277,31 +301,33 @@ func (o *operator) resources(ctx context.Context) ([]kruntime.Object, error) {
 		// covers the case of an admin-disable, we need to update dnsmasq on each node
 		cluster.Spec.GatewayDomains = make([]string, 0)
 	}
+	return cluster, nil
+}
 
-	// create a secret here for genevalogging, later we will copy it to
-	// the genevalogging namespace.
-	return append(results,
-		&corev1.Secret{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      pkgoperator.SecretName,
-				Namespace: pkgoperator.Namespace,
-			},
-			Data: map[string][]byte{
-				genevalogging.GenevaCertName: gcsCertBytes,
-				genevalogging.GenevaKeyName:  gcsKeyBytes,
-				corev1.DockerConfigJsonKey:   []byte(ps),
-			},
-		},
-		cluster,
-	), nil
+func (o *operator) SyncClusterObject(ctx context.Context) error {
+	resource, err := o.clusterObject(ctx)
+	if err != nil {
+		return err
+	}
+	return o.dh.Ensure(ctx, resource)
 }
 
-func (o *operator) CreateOrUpdate(ctx context.Context) error {
+func (o *operator) CreateOrUpdate(ctx context.Context, isInstall bool) error {
```

Comment:
I think we should split this into 2 separate functions: Create and Update with a common internal method.

Reply:
done